src/atomWithHash.ts

```diff
@@ -15,16 +15,52 @@ const safeJSONParse = (initialValue: unknown) => (str: string) => {
   }
 };
 
+export type SetHashOption =
+  | 'default'
+  | 'replaceState'
+  | ((searchParams: string) => void);
+
+export type AtomWithHashSetOptions = {
+  setHash?: SetHashOption;
+};
+
+export const setHashWithPush = (searchParams: string) => {
+  window.location.hash = searchParams;
+};
+
+export const setHashWithReplace = (searchParams: string): void => {
+  window.history.replaceState(
+    window.history.state,
+    '',
+    `${window.location.pathname}${window.location.search}#${searchParams}`,
+  );
+};
```

Comment:
Perhaps these two functions don't need to be exported if their just used here?

Reply:
i was thinking folks could make use of them as utils in their custom hash functions potentially but wdyt?

Reply:
i think this is the only remaining thing? are you ok exporting these or should i remove?

Reply:
Hmmm, I guess it's fine to merge it with the exports :ok_hand:

Reply:
Woohoo!!